src/mixins/observable.mixin.js

```diff
@@ -1,145 +0,0 @@
-    this.on(eventName, _handler);
-    return _handler;
-  }
-
-  /**
-   * Observes specified event **once**
-   * @memberOf fabric.Observable
-   * @alias once
-   * @param {String|Object} eventName Event name (eg. 'after:render') or object with key/value pairs (eg. {'after:render': handler, 'selection:cleared': handler})
-   * @param {Function} handler Function that receives a notification when an event of the specified type occurs
-   * @return {Function} disposer
-   */
-  function once(eventName, handler) {
-    // one object with key/value pairs was passed
-    if (arguments.length === 1) {
-      var handlers = {};
-      for (var prop in eventName) {
-        handlers[prop] = _once.call(this, prop, eventName[prop]);
-      }
-      return off.bind(this, handlers);
-    }
-    else {
-      var _handler = _once.call(this, eventName, handler);
-      return off.bind(this, eventName, _handler);
-    }
-  }
-
-  /**
-   * Stops event observing for a particular event handler. Calling this method
-   * without arguments removes all handlers for all events
-   * @memberOf fabric.Observable
-   * @alias off
-   * @param {String|Object} eventName Event name (eg. 'after:render') or object with key/value pairs (eg. {'after:render': handler, 'selection:cleared': handler})
-   * @param {Function} handler Function to be deleted from EventListeners
-   */
-  function off(eventName, handler) {
-    if (!this.__eventListeners) {
-      return;
-    }
-
-    // remove all key/value pairs (event name -> event handler)
-    if (arguments.length === 0) {
-      for (eventName in this.__eventListeners) {
-        _removeEventListener.call(this, eventName);
-      }
-    }
-    // one object with key/value pairs was passed
-    else if (typeof eventName === 'object' && typeof handler === 'undefined') {
-      for (var prop in eventName) {
-        _removeEventListener.call(this, prop, eventName[prop]);
-      }
-    }
-    else {
-      _removeEventListener.call(this, eventName, handler);
-    }
-  }
-
-  /**
-   * Fires event with an optional options object
-   * @memberOf fabric.Observable
-   * @param {String} eventName Event name to fire
-   * @param {Object} [options] Options object
-   */
-  function fire(eventName, options) {
-    if (!this.__eventListeners) {
-      return;
-    }
-
-    var listenersForEvent = this.__eventListeners[eventName];
-    if (!listenersForEvent) {
-      return;
-    }
-
-    for (var i = 0, len = listenersForEvent.length; i < len; i++) {
-      listenersForEvent[i] && listenersForEvent[i].call(this, options || { });
-    }
-    this.__eventListeners[eventName] = listenersForEvent.filter(function(value) {
-      return value !== false;
-    });
-  }
-
-  /**
-   * @namespace fabric.Observable
-   * @tutorial {@link http://fabricjs.com/fabric-intro-part-2#events}
-   * @see {@link http://fabricjs.com/events|Events demo}
-   */
-  fabric.Observable = {
-    fire: fire,
-    on: on,
-    once: once,
-    off: off,
-  };
```

Comment:
Observable was an object with 4 methods. Why does to become a class?

Reply:
my intention is to make CommonMethods extend Observable